server/src/yml-objects/AbstractYmlObject.ts

```diff
@@ -82,26 +86,58 @@ export abstract class AbstractYmlObject implements CompletionItem {
         scopeStartOffset?: number,
         scopeEndOffset?: number,
     ): void {
-        sourceElementName = sourceElementName ? sourceElementName : 'static';
+        this.sourceElementName = sourceElementName ? sourceElementName : 'STATIC';
         this.data = `id_${sourceElementName}_${this.label}`;
-        this.setDocumentation(getDocumentation(fields, connection));
-        this.detail = getType(fields, connection, baseType);
+        const doc = getDocumentation(fields, connection);
+        this.setDetail(getType(fields, connection, baseType));
+        this.setUserInformations(this.detail, doc);
         if (scopeEndOffset && scopeStartOffset) {
             this.scopeEndOffset = scopeEndOffset;
             this.scopeStartOffset = scopeStartOffset;
         }
     }
 
+    public setDetail(type: string): void {
+        this.detail =
+            this.sourceElementName === 'STATIC'
+                ? `(${this.kindName}) [${this.sourceElementName}] ${this.label} ⇒ ${type}`
+                : `(${this.kindName}) [${this.sourceElementName}].${this.label} ⇒ ${type}`;
+    }
+
+    public getHoverContent(): MarkupContent {
```

Comment:
It’s a (tiny) pity this is recomputed each time, but I dunno what’s best between this and having to update it in every relevant setter. :thinking: 

```java
private foo
private bar
private fooBar

constructor(…) {
  …
  updateFooBar()
}

getFooBar() {
  return fooBar
}

setFoo(x) {
  foo = x
  updateFooBar()
}

setBar(x) {
  bar = x
  updateFooBar()
}

private updateFooBar() {
  fooBar = foo + bar
}
```

LOL I’m soooo overthinking this!

Reply:
Well, in fact, once I'm sure I got both documentation and detail, I can save this as constant, since I recreate everything when there is a change in the file.